### PR TITLE
docs(headless): add React sample for headless relevance inspector

### PR DESCRIPTION
**Pull request by btaillon**
https://coveord.atlassian.net/browse/KIT-423

The checkbox doesn't work without https://github.com/coveo/ui-kit/pull/530.

### CODE REVIEW

File: packages/samples/headless-react/src/components/relevance-inspector/relevance-inspector.class.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+  private updateState() {
+    this.setState(this.controller.state);
+  }
+
+  render() {
+    return (
+      <div>
+        <label>
+          Debugging:{' '}
```
Comment (olamothe) [inline packages/samples/headless-react/src/components/relevance-inspector/relevance-inspector.class.tsx:42]: Maybe `Enable debug mode` ?

**Review by olamothe — APPROVED** [review]

**Review by ThibodeauJF — APPROVED** [review]

**Review by nrawji — APPROVED** [review]

File: packages/samples/headless-react/src/components/relevance-inspector/relevance-inspector.fn.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+  return (
+    <div>
+      <label>
+        Enable debug mode:{' '}
+        <input
+          type="checkbox"
+          checked={state.isEnabled}
+          onChange={() =>
+            state.isEnabled ? controller.disable() : controller.enable()
```
Comment (samisayegh) [inline packages/samples/headless-react/src/components/relevance-inspector/relevance-inspector.fn.tsx:35]: Is the condition order correct here? When `isEnabled` is `true`, we disable the relevance inspector? Shouldn't it be the opposite?

**Review by samisayegh — APPROVED** [review]

File: packages/samples/headless-react/src/components/relevance-inspector/relevance-inspector.fn.tsx (hunk 2 above)
Comment (btaillon) [reply packages/samples/headless-react/src/components/relevance-inspector/relevance-inspector.fn.tsx:35]: Yes, the order is correct, I tested it.

Comment (samisayegh) [reply packages/samples/headless-react/src/components/relevance-inspector/relevance-inspector.fn.tsx:35]: I guess it just reads weird then? If the checkbox is enabled, we `disable` the inspector? 🤔

Comment (btaillon) [reply packages/samples/headless-react/src/components/relevance-inspector/relevance-inspector.fn.tsx:35]: when `onChange` is called, `state.isEnabled` isn't changed until `controller.disable()` or `controller.enable()` is called, which triggers a re-render.

**Pull request merged**